lib/Cake/Model/Datasource/DboSource.php

```diff
@@ -1162,7 +1160,20 @@ public function read(Model $Model, $queryData = array(), $recursive = null) {
 					continue;
 				}
 
-				if ($noAssocFields) {
+				if (!empty($queryData['fields'])) {
+					foreach ($queryData['fields'] as $key => $field) {
+						$fieldExploded = explode('.', $field);
+						if (count($fieldExploded) === 2 && str_replace('`', '', $fieldExploded[0]) === $LinkModel->alias) {
```

Comment:
The backticks here should be `$this->startQuote` and `$this->endQuote`. Unfortunately the quotes are not the same across all the supported databases. For example SQLServer uses `[` and `]`.